gitops/vcs/github/github.go

```diff
@@ -0,0 +1,47 @@
+package github
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"golang.org/x/oauth2"
+	"github.com/google/go-github/v32/github"
+)
+
+var (
+	repoOwner = flag.String("github_repo_owner", "", "the owner user/organization to use for github api requests")
+	repo = flag.String("github_repo", "", "the repo to use for github api requests")
+	pat = flag.String("github_access_token", "", "the access token to authenticate requests")
```

Comment:
I'd do  `os.Getenv("GITHUB_TOKEN")` as default value here. https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#about-the-github_token-secret

Reply:
good idea

Reply:
I think it is also good idea to add `github_api_endpoint` parameter that defaults to public GitHub.